Artifacts/windows-deploymentgroup/InstallAndConfigureDeploymentGroupAgent.ps1

```diff
@@ -144,6 +144,76 @@ function Configure-DeploymentGroupAgent
 
     if ($runAsAutoLogon)
     {
+        # Create a PS session for the user to trigger the creation of the registry entries required for autologon
+        if ($windowsLogonAccount.Split("\").Count -eq 2)
+        {
+            $domain = $windowsLogonAccount.Split("\")[0]
+        }
+        else
+        {
+            $domain = Hostname
+        }
+        $computerName = "localhost"
+        $password = ConvertTo-SecureString $windowsLogonPassword -AsPlainText -Force
+        $credentials = New-Object System.Management.Automation.PSCredential("$domain\\$windowsLogonAccount", $password)
+        Enter-PSSession -ComputerName $computerName -Credential $credentials
+        Exit-PSSession
+
+        $ErrorActionPreference = "stop"
+
+        # 120 seconds timeout
+        $timeout = 120 
+
+        try
+        {
+            # Check if the HKU drive already exists
+            Get-PSDrive -PSProvider Registry -Name HKU | Out-Null
+            $canCheckRegistry = $true
+        }
+        catch [System.Management.Automation.DriveNotFoundException]
+        {
+            try 
+            {
+            # Create the HKU drive
+            New-PSDrive -PSProvider Registry -Name HKU -Root HKEY_USERS | Out-Null
+            $canCheckRegistry = $true
+            }
+            catch 
+            {
+            # Ignore the failure to create the drive and go ahead with trying to set the agent up
+            Write-Warning "Moving ahead with agent setup as the script failed to create HKU drive necessary for checking if the registry entry for the user's SId exists.\n$_"
+            }
+        }
+
+        # Check if the registry key required for enabling autologon is present on the machine, if not wait for 120 seconds in case the user profile is still getting created
+        while ($timeout -ge 0 -and $canCheckRegistry)
+        {
+            $objUser = New-Object System.Security.Principal.NTAccount($windowsLogonAccount)
```

Comment:
Should you put all this logic in a `try`/`catch`? Because you have `$ErrorActionPreference = "stop"`, any issues that happen here will force the script to stop. It may not be what you want.

Reply:
Actually no. There is a crucial step New-Item -Path "HKU:\\$securityId\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run" -Force, If this fails then the agent will get registered to tfs but won't start up. I would want the script to stop if this check or this creation fails. 

The only case in which I go ahead is if in case i somehow do not have permissions to check for this key then I do not have sufficient info to decide whether to stop or go ahead.

case in point being the previous catch block.